### PR TITLE
Image loading on upload

### PR DESCRIPTION
**Pull request by SaraVieira**
This PR adds a loading for an image is being uploaded: https://github.com/webstudio-is/webstudio-designer/issues/37

## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [ ] made a self-review
- [x] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [ ] conceptual
  - [ ] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [ ] tested locally and on preview environment
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the 
- [ ] `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/designer/features/sidebar-left/panels/asset-manager.tsx [hunk 1]
```diff
@@ -1,44 +1,49 @@
...
+        />
       </Flex>
       <Grid columns={2} gap={2}>
         {assets.map((asset) => (
-          <AssetManagerImage key={asset.id} asset={asset} />
+          <AssetManagerImage
+            key={asset.id}
+            path={asset.path}
+            alt={asset.alt || asset.name}
+            uploading={"uploading" in asset && asset.uploading}
```
Comment (kof) [inline apps/designer/app/designer/features/sidebar-left/panels/asset-manager.tsx:45]: shouldn't it be asset.state or something with uploading one of the states and to be always defined?

Comment (kof) [reply apps/designer/app/designer/features/sidebar-left/panels/asset-manager.tsx:45]: or asset.status

File: apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx [hunk 2]
```diff
@@ -18,21 +17,60 @@ const useImageWithFallback = ({ path }: { path: string }) => {
...
+  path: string;
+  alt?: string;
+  uploading?: boolean;
 }) => {
   const src = useImageWithFallback({ path });
+  const [progressBarPercentage, setProgressBarPercentage] = useState(0);
+
+  useEffect(() => {
+    if (uploading) {
+      setInterval(() => {
```
Comment (kof) [inline apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx:33]: missing a cleanup, also react-use has this in a cleaned up version, maybe its better to use it?

Comment (kof) [reply apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx:33]: danger is that user closes, component unmounts, this still still runs and results in warnings from react

File: apps/designer/app/designer/features/sidebar-left/panels/asset-manager.tsx (hunk 1 above)
Comment (SaraVieira) [reply apps/designer/app/designer/features/sidebar-left/panels/asset-manager.tsx]: Fixed this and used ststus

File: apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx (hunk 2 above)
Comment (SaraVieira) [reply apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx]: also done!

File: apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx [hunk 3]
```diff
@@ -18,21 +19,60 @@ const useImageWithFallback = ({ path }: { path: string }) => {
...
+  path,
+  alt,
+  status,
 }: {
-  asset: Asset;
+  path: string;
+  alt?: string;
+  status?: Asset["status"];
 }) => {
+  const uploading = status === "uploading";
```
Comment (kof) [inline apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx:30]:
```suggestion
  const isUploading = status === "uploading";
```

File: apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx [hunk 4]
```diff
@@ -18,21 +19,60 @@ const useImageWithFallback = ({ path }: { path: string }) => {
...
   const src = useImageWithFallback({ path });
+  const [progressBarPercentage, setProgressBarPercentage] = useState(0);
+
+  useInterval(
```
Comment (kof) [inline apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx:34]:
```suggestion
  @todo rewrite this fake indication to show real progress
  useInterval(
```

**Review by kof — APPROVED** [review]
found a few nitpicks, otherwise 👍

File: apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx [hunk 5]
```diff
@@ -18,21 +19,61 @@ const useImageWithFallback = ({ path }: { path: string }) => {
...
-  asset: Asset;
+  path: string;
+  alt?: string;
+  status?: Asset["status"];
 }) => {
+  const uploading = status === "uploading";
   const src = useImageWithFallback({ path });
+  const [progressBarPercentage, setProgressBarPercentage] = useState(0);
+
+  @todo rewrite this fake indication to show real progress
```
Comment (kof) [inline apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx:34]: sorry
```suggestion
  // @todo rewrite this fake indication to show real progress
```

File: apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx (hunk 3 above)
Comment (SaraVieira) [reply apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx]: Fixed

**Pull request merged**